From a88c2b7af91d4e992df967e90db2a49f529ccb88 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:56 -0400
Subject: [ethernet] cxgb4: Call dev_kfree/consume_skb_any instead of [dev_]kfree_skb

Message-id: <1410475447-94392-38-git-send-email-svemuri@redhat.com>
Patchwork-id: 93874
O-Subject: [RHEL7.1 PATCH BZ 1124947 037/108] cxgb4: Call dev_kfree/consume_skb_any instead of [dev_]kfree_skb.
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Replace kfree_skb with dev_consume_skb_any in free_tx_desc that can be
called in hard irq and other contexts. dev_consume_skb_any is used
as this function consumes successfully transmitted skbs.

Replace dev_kfree_skb with dev_kfree_skb_any in t4_eth_xmit that can
be called in hard irq and other contexts, on paths that drop the skb.

Replace dev_kfree_skb with dev_consume_skb_any in t4_eth_xmit that can
be called in hard irq and other contexts, on paths that successfully
transmit the skb.

Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
(cherry picked from commit a7525198a85102034a442fba75352754bd439c2b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/sge.c b/drivers/net/ethernet/chelsio/cxgb4/sge.c
index 9e15ce2..1564b0a 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/sge.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/sge.c
@@ -1009,7 +1009,7 @@ netdev_tx_t t4_eth_xmit(struct sk_buff *skb, struct net_device *dev)
   * anything shorter than an Ethernet header.
   */
  if (unlikely(skb->len < ETH_HLEN)) {
-out_free: dev_kfree_skb(skb);
+out_free: dev_kfree_skb_any(skb);
   return NETDEV_TX_OK;
  }
 
@@ -1104,7 +1104,7 @@ out_free: dev_kfree_skb(skb);
 
  if (immediate) {
   inline_tx_skb(skb, &q->q, cpl + 1);
-  dev_kfree_skb(skb);
+  consume_skb(skb);
  } else {
   int last_desc;
 
-- 
1.7.1